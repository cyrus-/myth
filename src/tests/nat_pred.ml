let s = Nat_decls.s ^ "

let nat_pred : nat -> nat |>
  { O => O
  ; S (O) => O
  ; S (S (O)) => S (O) } = ?"
