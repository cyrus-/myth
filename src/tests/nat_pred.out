(* Generated Thu Nov 13 00:35:52 2014 *)
let nat_pred : nat -> nat =
  fun (n1:nat) -> match n1 with
                    | O -> O
                    | S (n2) -> n2
;;
